src/ck-ldb/pheap.h

```diff
@@ -0,0 +1,131 @@
+#ifndef PROC_HEAP_H
+#define PROC_HEAP_H
+
+#include <vector>
+
+// custom heap to allow removal of processors from any position
```

Comment:
Instead of this custom class, could you use a `std::multiset`?

Reply:
I need a heap, not a fully sorted container. But I guess it really comes down to which one is faster in practice (I haven't tried multiset).

Reply:
I think it's worth keeping @juanjgalvez's implementation because of the constant time element lookup (at the cost of extra storage), at least for now. I haven't benchmarked it, but I will in the future (there are other places where non-standard data structures are used, I think this should be lumped in with the effort of checking if those should be replaced).